### DISCUSSION TITLE
时间戳timestamp 2038年问题

### DISCUSSION THREAD

**Discussion by Mir-YJR** [taosdata/TDengine#4945] [General]
目前的时间戳有最大2038年限制，能不能通过设置成单位为s去解决这个隐患呢。

**Comment by sangshuduo** [comment]
保存的是 Posix Timestamp

**Reply by Mir-YJR** [reply]
但是插入的是毫秒，插入大于2038年的数据就会报错。如何插入一个大于2038年的数据呢？

**Comment by sangshuduo** [comment]
时序数据是处理真实的物联网传感器数据，为什么要在2021年插2038年的数据？

**Reply by Mir-YJR** [reply]
前瞻性考虑，到了2038年怎么办，如果那时候，原有数据很庞大了，数据迁移？

**Comment by sangshuduo** [comment]
后面版本会规划如何支持
